pg_search/src/api/aggregate.rs

```diff
@@ -42,3 +43,41 @@ pub fn aggregate(
         bucket_limit.try_into()?,
     )?))
 }
+
+/// State transition function for agg aggregate
+/// This accumulates the agg definition (which should be the same across all rows)
+#[pg_extern(stable, parallel_safe)]
+pub fn agg_sfunc(state: Option<Internal>, agg_definition: JsonB) -> Option<Internal> {
+    // On first call, validate and store the agg definition
+    if state.is_none() {
+        let agg_value: serde_json::Value = agg_definition.0;
+
+        // Validate it's a valid Tantivy aggregation by attempting to deserialize
+        if let Err(e) = serde_json::from_value::<Aggregation>(agg_value.clone()) {
+            pgrx::error!("Invalid Tantivy aggregation definition: {}", e);
+        }
+
+        // Store the validated JSON in an Internal datum
+        // We use a Box<serde_json::Value> to store it
+        unsafe {
+            let boxed = Box::new(agg_value);
+            Some(
+                pgrx::Internal::from_datum(
+                    pgrx::pg_sys::Datum::from(Box::into_raw(boxed) as *mut std::ffi::c_void),
+                    false,
+                )
+                .unwrap(),
+            )
+        }
+    } else {
+        // Just return the existing state
+        state
+    }
+}
```

Comment:
This is a placeholder like `paradedb.score`, right? So it shouldn't actually run, and instead will be projected by the custom scan?

If that's the case, it would be good to eagerly fail here if this does run, I think.

But additionally: note that `pgrx` has native support for aggregate functions: I added one in https://github.com/paradedb/paradedb/pull/3336. It might be good to use that same support here?

Reply:
Thanks for the suggestion. Fixed this in 0f9a44d490f0f602fbd82b80b4d8897070c881e0